gssapi/tests/test_raw.py

```diff
@@ -772,6 +772,34 @@ def test_sasl_names(self):
             cmp_mech.shouldnt_be_none()
             cmp_mech.should_be(mech)
 
+    @ktu.gssapi_extension_test('rfc4178', 'Negotiation Mechanism')
+    def test_set_neg_mechs(self):
```

Comment:
While I think we want to enforce mech_set not being None (even though 4178 doesn't require it), I would like to see us test mech selection.  We should have the NTLM mechanism around - maybe something using that?  (See test_set_sec_context_option and test_set_sec_context_option_fail.)  You'll need to install gssntlmssp to check this locally.